README.md

```diff
@@ -1,8 +1,200 @@
-playbooks to resolve the root cause.
+This project contains the _auto-heal_ service. It receives alert notifications
+from the Prometheus alert manager and executes Ansible playbooks to resolve the
+root cause.
+
+## Configuration
+
+Most of the configuration of the auto-heal service is kept in a YAML
+configuration file.  The name of the configuration file is specified using the
+`-config-file` command line option. If this option isn't explicitly given then
+the service will try to load the `autoheal.yml` file from the current working
+directory.
+
+In addition to the configuration file the auto-heal service also uses command
+line options to configure the connection to the Kubernetes API and the log
+level. Use the `-h` option to get a complete list of these command line options.
+
+The `-kubeconfig` command line option is used to specify the location of the
+Kubernetes client configuration file. When running outside of a Kubernetes
+cluster the auto-heal service will use `$HOME/.kube/config` by default, the same
+used by the `kubectl` command. When running inside a Kubernetes cluster it will
+use the configuration that Kubernetes mounts automatically in the pod file
+system. So in most cases this command line option won't have to be explicitly
+included.
+
+The `-alsologtostderr` option is very convenient when running the auto-heal
+service, both in development and production environments.
+
+Assuming that you want to have your own `my.yml` configuration file a typical
+command line will be the following:
+
+```bash
+$ autoheal -config-file my.yml -alsologtostderr
+```
+
+See the `autoheal.yml` file for a complete example.
+
+### AWX or AnsibleTower configuration
+
+The first section of the configuration file is named `awx` and it contains all
+the details needed to connect to the [AWX](https://www.ansible.com/products/awx-project)
+or [Ansible Tower](https://www.ansible.com/products/tower) server:
+
+```yaml
+awx:
+  address: https://myawx.example.com/api
+  proxy: http://myproxy.example.com:3128
+  credentialsRef:
+    namespace: my-namespace
+    name: my-awx-credentials
+  tlsRef:
+    namespace: my-namespace
+    name: my-awx-ca
+  project: "Auto-heal"
+```
+
+The `address` parameter is the URL of the API of the AWX server. It should
+contain the `/api` suffix, but not the `/v1` or `/v2` suffix, as the auto-heal
+service will internally decide which version to use.
+
+The `proxy` parameter is optional, and it indicates what HTTP proxy should be
+used to connect to the AWX API. If this parameter is not specified, or if it is
+empty, then the connection will be direct to the AWX server, without a proxy.
+
+The `credentialsRef` parameter is a reference to the [Kubernetes
+secret](https://kubernetes.io/docs/concepts/configuration/secret) that contains
+the user name and password used to connect to the AWX API. That secret should
+contain the `username` and `password` keys. For example:
```

Comment:
consider replacing the above section with a link to autoheal.yml so we have to maintain only one file.